### PR TITLE
curvefs: fix trash will delete file data more than once

### PR DESCRIPTION
**Pull request by SeanHai**
### What problem does this PR solve?

#2774 

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by SeanHai** [comment]
cicheck

File: curvefs/src/metaserver/trash.cpp [hunk 1]
```diff
@@ -52,27 +54,38 @@ void TrashImpl::Init(const TrashOption &option) {
...
+    VLOG(6) << "Add Trash Item success, fsId = " << fsId_
+            << ", partitionId = " << partitionId_
+            << ", inodeId = " << inodeId
+            << ", dtime = " << dtime;
 }
 
 void TrashImpl::ScanTrash() {
     LockGuard lgScan(scanMutex_);
+    // only scan on leader
+    if (copysetNode_ == nullptr || !copysetNode_->IsLeaderTerm()) {
```
Comment (xu-chaojie) [inline curvefs/src/metaserver/trash.cpp:85]: The following cycle also needs to be interrupted when the node is not the leader.

File: curvefs/src/metaserver/trash.cpp [hunk 2]
```diff
@@ -85,19 +98,16 @@ void TrashImpl::ScanTrash() {
...
                            << ", ret = " << MetaStatusCode_Name(ret);
                 it++;
                 continue;
             }
-            VLOG(6) << "Trash Delete Inode, fsId = " << it->fsId
-                    << ", inodeId = " << it->inodeId;
+            LOG(INFO) << "Trash delete inode, fsId = " << fsId_
+                      << ", partitionId = " << partitionId_
+                      << ", inodeId = " << it->inodeId;
             it = temp.erase(it);
```
Comment (xu-chaojie) [inline curvefs/src/metaserver/trash.cpp:111]: How to remove trash items on the follower side

File: curvefs/src/metaserver/trash.cpp [hunk 3]
```diff
@@ -219,39 +209,49 @@ MetaStatusCode TrashImpl::DeleteInodeAndData(const TrashItem &item) {
...
+    DeleteInodeRequest request;
+    request.set_poolid(poolId_);
+    request.set_copysetid(copysetId_);
+    request.set_partitionid(partitionId_);
+    request.set_fsid(fsId_);
+    request.set_inodeid(item.inodeId);
+
+    DeleteInodeResponse response;
+    DeleteInodeClosure done;
+    auto DeleteInodeOp = new copyset::DeleteInodeOperator(
```
[the remark this answers is not in the document]
Comment (SeanHai) [reply curvefs/src/metaserver/trash.cpp:248]: This operation originally existed and was used to delete the remaining inodes after the creation process failed.

https://github.com/opencurve/curve/blob/dd8ce61654940a8e789960ce5c03c1c84b09bf0a/curvefs/src/metaserver/copyset/meta_operator.h#L359

**Comment by SeanHai** [comment]
cicheck

**Review by wuhongsong — APPROVED** [review]

**Review by xu-chaojie — APPROVED** [review]

**Pull request merged**